test/net/sourceforge/kolmafia/request/AscensionRequestTest.java

```diff
@@ -0,0 +1,143 @@
+package net.sourceforge.kolmafia.request;
+
+import static internal.helpers.Networking.html;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AscensionRequestTest {
+  @BeforeEach
+  public void initializeCharPrefs() {
+    KoLCharacter.reset("fakePrefUser");
+    KoLCharacter.reset(true);
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @AfterEach
+  public void resetCharAndPrefs() {
+    KoLCharacter.reset("");
+    KoLCharacter.reset(true);
+    KoLCharacter.setUserId(0);
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @Test
+  void testAscensionsTodayTracked() {
+    // Simple test to confirm that it resets our last run breakfast
+    String breakfastPref = "lastBreakfast";
+    int breakfastBefore = 5;
+    int breakfastAfter = 0;
+    int breakfastValhalla = -1;
+    String ascensionsPref = "ascensionsToday";
+    int ascensionBefore = 1;
+    int ascensionAfter = 2;
+
+    // Set breakfast and ascension prefs
+    Preferences.setInteger(ascensionsPref, ascensionBefore);
```

Comment:
These should all be withProperty cleanups

Reply:
Changed